src/Model.jl

```diff
@@ -1,5 +1,21 @@
 # Contains the physical specification of the model
 
+using Unitful
+import Unitful.uconvert
+using UnitfulAtomic
+
+# Convert a temperature to an energy via the Boltzmann constant.
+function uconvert(a::Unitful.EnergyUnits, x::Unitful.Quantity{T,Unitful.𝚯,U}) where {T <: Real, U <: Unitful.Units}
+    uconvert(a, Unitful.k * x)
+end
+
+# Convert a value to an energy in atomic units.
+# If given a number, assume it is already in atomic units (pass through directly).
+function to_energy(x::Unitful.Quantity)::Real
```

Comment:
Here annotating the `::Real` is actually pretty bad. The reason is that if you pass the function a concrete type (let's say a float annotated by the unit) the compiler will infer that the return value is a Float64 and directly compile the function that way. If you however annotate the return type as `::Real` (where `Real` is a supertype), than this *forces* to return the generic type, which in general a lot slower, since it cannot be stored on the stack for example. In this case it should not matter, but there is no reason to put it there, so just don't. That's less writing *and* faster code :smile:.

When to use generic types and when not one needs a little experience, but for starters (especially for people coming from the C++ world like you seem to): Don't put types and only put them if the compiler complains or you really need to!